src/freenet/node/updater/MainJarDependenciesChecker.java

```diff
@@ -1576,6 +1601,97 @@ private boolean createRunShNoNice(File input, File output) throws IOException {
         }
 
 	}
+	
+	/** Deploys a multi-file replace on Windows with a restart, using a simple cmd script.
+	 * Unlike on Unix, we cannot actually move the files first, the script must do that. */
+    private class WindowsRestartingAtomicDeployer extends RestartingAtomicDeployer {
+
+        public WindowsRestartingAtomicDeployer(String name) {
+            super(name);
+        }
+        
+        static final String RESTART_SCRIPT_NAME = "tempRestartFreenet.cmd";
+        
+        @Override
+        protected boolean deployMultiFileUpdate() {
+            if(!WrapperManager.isControlledByNativeWrapper()) return false;
+            File restartScript;
+            try {
+                restartScript = createRestartScript();
+            } catch (IOException e) {
+                System.err.println("Unable to deploy multi-file update for "+name+" because cannot write script to restart the wrapper: "+e);
+                Logger.error(this, "Unable to deploy multi-file update for "+name+" because cannot write script to restart the wrapper: "+e, e);
+                return false;
+            }
+            if(restartScript == null) return false;
+            try {
+                ProcessBuilder pb = new ProcessBuilder("cmd","/c",restartScript.getAbsolutePath());
+                pb.directory(deployer.getNodeDir());
+                pb.redirectErrorStream(true);
+                pb.redirectError(new File(deployer.getNodeDir(), "update-script.log"));
+                pb.redirectInput(new File("NUL:"));
+                if(pb.start() == null) return false;
+            } catch (IOException e) {
+                System.err.println("Unable to start restarter script "+restartScript+" -> cannot deploy multi-file update for "+name+" : "+e);
+                Logger.error(this, "Unable to start restarter script "+restartScript+" -> cannot deploy multi-file update for "+name+" : "+e, e);
+                return false;
+            }
+            System.out.println("Shutting down Freenet for hard restart after deploying multi-file update for "+name+". The script "+restartScript+" should start it back up.");
+            WrapperManager.stop(0);
+            return true;
+        }
+        
+        private File createRestartScript() throws IOException {
+            File restartScript = new File(deployer.getNodeDir(), RESTART_SCRIPT_NAME);
+            restartScript.delete();
+            FileBucket fb = new FileBucket(restartScript, false, true, false, false);
+            OutputStream os = null;
+            try {
+                os = new BufferedOutputStream(fb.getOutputStream());
+                OutputStreamWriter osw = new OutputStreamWriter(os); // Use default locale.
```

Comment:
I'd really prefer that this script be a resource file - perhaps in two parts if we really need to generate which files to move - instead of writing lots of individual lines. Thoughts?